ethcore/src/client/client.rs

```diff
@@ -1652,6 +1652,33 @@ impl MiningBlockChainClient for Client {
 		open_block
 	}
 
+	fn reopen_block(&self, block: ClosedBlock) -> OpenBlock {
+		let engine = &*self.engine;
+		let mut block = block.reopen(engine);
+		let max_uncles = engine.maximum_uncle_count();
+		if block.uncles().len() < max_uncles {
+			let chain = self.chain.read();
+			let h = chain.best_block_hash();
+			// Add new uncles
+			let uncles = chain
+				.find_uncle_hashes(&h, engine.maximum_uncle_age())
+				.unwrap_or_else(Vec::new);
+
+			for h in uncles {
```

Comment:
`prepare_open_block` uses `find_uncle_headers` and not `find_uncle_hashes`. Borrow checker won't allow using a single iterator sequence here because `blocks` would need to be used in two closures.